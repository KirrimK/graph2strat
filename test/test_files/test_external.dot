#init Init
digraph "self.test" {
    Init [comment="self.on_enter/self.on_leave"] //bruh
    Outhome [comment="self.on_enter_/self.on_leave_"]
    {Init Outhome} -> End [label="self.on_transition/self.guard"]
}