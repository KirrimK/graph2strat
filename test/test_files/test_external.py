# importations

class TestTemplating():

    # attributs de classe

    def __init__(self):
        # initialisation des attributs, super(), etc...

        # insertion des états et transitions
        """STATES_FILE:./test_external.dot"""

        # autre code d'__init__
        pass

    # définitions de méthodes
    
    # définitions des gardes et transitions à ne pas écraser
    def ma_garde(self):
        return False

    def on_leave(self):
        pass

    # insertion des placeholders de gardes, actions de transitions
    """HANDLERSPL_BEGIN

    IGNORE self.ma_garde
    IGNORE self.on_leave
    IGNORE self.on_enter
    
    HANDLERSPL_END"""

def main(args=None):
    strat = TestTemplating()
    # autre code

if __name__ == '__main__':
    main()
