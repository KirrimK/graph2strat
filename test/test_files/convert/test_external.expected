# File generated using graph2strat by KirrimK@ENAC vunknown
# Don't forget to import the contents of statemachine.py: State, Transition, StateMachine
# importations

class TestTemplating():

    # attributs de classe

    def __init__(self):
        # initialisation des attributs, super(), etc...

        # insertion des états et transitions
        # [graph2strat generated states and transitions]
        Outhome = State("Outhome", self.on_enter_, self.on_leave_)
        Init = State("Init", self.on_enter, self.on_leave)
        End = State("End")
        InitOuthomeToEnd = Transition("InitOuthomeToEnd", End, self.on_transition, self.guard)
        Init.add_transition(InitOuthomeToEnd)
        Outhome.add_transition(InitOuthomeToEnd)
        self.test = StateMachine(Init)
        # [end of generated content]

        # autre code d'__init__
        pass

    # définitions de méthodes
    
    # définitions des gardes et transitions à ne pas écraser
    def ma_garde(self):
        return False

    def on_leave(self):
        pass

    # insertion des placeholders de gardes, actions de transitions
    # [graph2strat generated placeholder handlers]
    def guard(self):
        return True

    def on_transition(self):
        pass

    def on_leave_(self):
        pass

    def on_enter_(self):
        pass

    # [end of generated content]

def main(args=None):
    strat = TestTemplating()
    # autre code

if __name__ == '__main__':
    main()
